src/Withdrawal.java

```diff
@@ -116,8 +116,13 @@ private int displayMenuOfAmounts()
             case 2: // (i.e., chose option 1, 2, 3, 4 or 5), return the
             case 3: // corresponding amount from amounts array
             case 4:
-            case 5:
                userChoice = amounts[ input ]; // save user's choice
+               break;
+            case 5:
+               screen.displayMessage( "\nPlease insert the desired amount: " );
+               int customAmount = keypad.getInput(); // get user input through keypad
+            
+               userChoice = customAmount;
                break;       
             case CANCELED: // the user chose to cancel
                userChoice = CANCELED; // save user's choice
```

Comment:
Re-use an existing variable in a different context is not a best practice. In this case, userChoice is used also to store the amount. This could potentially cause unexpected behaviour. And in fact, with this implementation you cannot withdraw 6 dollars, because it’s the same number as the userChoice CANCELED

Reply:
Good catch. I fixed it. I dediced to return an invalid amount, instead of the CANCELED constant. I also changed the variable name.